### DISCUSSION TITLE
Is the server running on that host and accepting TCP/IP connections?

### DISCUSSION THREAD

**Discussion by Slimly-foutyfour** [netbox-community/netbox-docker#880] [Getting Help]
Am on my first build of netbox-docker community and my first posting - I need some help getting netbox-docker running. 

I am running
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy

docker -v
Docker version 20.10.12, build 20.10.12-0ubuntu4

docker-compose -v
docker-compose version 1.29.2, build unknown

contents of docker-compose.overide.yml

version: '3.4'
services:
  netbox:
   ports:
     - 8000:8080

during the docker-compose up 
postgres_1             | 2022-11-03 10:45:19.793 UTC [1] LOG:  starting PostgreSQL 14.5 on x86_64-pc-linux-musl, compiled by gcc (Alpine 11.2.1_git20220219) 11.2.1 20220219, 64-bit
postgres_1             | 2022-11-03 10:45:19.794 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1             | 2022-11-03 10:45:19.795 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1             | 2022-11-03 10:45:19.801 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1             | 2022-11-03 10:45:19.814 UTC [50] LOG:  database system was shut down at 2022-11-03 10:45:19 UTC
postgres_1             | 2022-11-03 10:45:19.834 UTC [1] LOG:  database system is ready to accept connections

but ends with 

netbox-worker_1        |     return worker_class(queues,
netbox-worker_1        |   File "/opt/netbox/venv/lib/python3.10/site-packages/rq/worker.py", line 219, in __init__
netbox-worker_1        |     connection.client_setname(self.name)
netbox-worker_1        |   File "/opt/netbox/venv/lib/python3.10/site-packages/redis/commands/core.py", line 697, in client_setname
netbox-worker_1        |     return self.execute_command("CLIENT SETNAME", name, **kwargs)
netbox-worker_1        |   File "/opt/netbox/venv/lib/python3.10/site-packages/redis/client.py", line 1235, in execute_command
netbox-worker_1        |     conn = self.connection or pool.get_connection(command_name, **options)
netbox-worker_1        |   File "/opt/netbox/venv/lib/python3.10/site-packages/redis/connection.py", line 1387, in get_connection
netbox-worker_1        |     connection.connect()
netbox-worker_1        |   File "/opt/netbox/venv/lib/python3.10/site-packages/redis/connection.py", line 615, in connect
netbox-worker_1        |     raise TimeoutError("Timeout connecting to server")
netbox-worker_1        | redis.exceptions.TimeoutError: Timeout connecting to server
netbox-docker_netbox-worker_1 exited with code 1
netbox_1               |        Is the server running on that host and accepting TCP/IP connections?
netbox_1               | [ Use DB_WAIT_DEBUG=1 in netbox.env to print full traceback for errors here ]
netbox_1               | ⏳ Waiting on DB... (0s / 30s)

am I missing something

**Comment by timrabl** [comment]
Your Redis container is not accessible. You can see that here:

```bash
...
netbox-worker_1 | File "/opt/netbox/venv/lib/python3.10/site-packages/redis/connection.py", line 615, in connect
netbox-worker_1 | raise TimeoutError("Timeout connecting to server")
```

Maybe there are already problems before when starting the Redis container ? Depending on where Redis runs with you.

But if you use the default compose file, then by default you should have a Redis and a Redis cache container in the same compose stack. If that is the case, then it sounds more like a problem with the Docker variables of the Redis / Redis Cache / NetBox instances or a Docker network problem.

But to my ears it sounds like a network problem. I assume that the container is down or not yet reachable when the netbox-worker container tries to connect to the redis database.
